WordPress/Learn
# Workshops: Add "series" taxonomy

Ostensibly this PR is about adding a "Workshop Series" taxonomy to associate certain workshop posts with each other. And the PR does do that. But it also updates a bunch of theme code under the hood and fixes a couple of tangentially related issues.

Series-related changes:

* Creates a "Workshop Series" taxonomy associated with the `wporg_workshop` CPT.
* Adds UI on single workshop pages to navigate to previous and next workshops in the series.
* Adds Workshop Series archive pages that resemble the Workshops post type archive, with a few minor differences:
  * No large featured workshop at the top.
  * Workshops appear in reverse order, so oldest first (since workshop order in a series is determined by publish date).

Other changes:

* Moves all taxonomy registration to the inc/taxonomy.php file.
* Moves all post type registration to the inc/post-type.php file.
* Homepage will now show 6 recent workshops.
* The Workshops archive page is de-duped so that the featured workshop does not also show up in the grid of other workshops.
* Adds pagination to the Workshops archive page.
* The large featured workshop component is only shown on Page 1.

Fixes #92
Fixes #94 
Fixes #104 

## Screenshots

Series navigation UI on a single workshop page:
![workshop-series_single](https://user-images.githubusercontent.com/916023/90928943-ac21d580-e3ac-11ea-93cf-8d6448e9aa13.jpg)

Series archive page:
![workshop-series_archive](https://user-images.githubusercontent.com/916023/90834033-25aeba80-e2fe-11ea-93d1-f21b2f9eb2a4.jpg)

## To test

1. Create a new Series term
1. Associate some workshop posts with it (preferably at least three)
1. View the single page for one of the workshops in the series. Note that you can navigate to the other workshops in the series.
1. Click on the name of the series on the single page to go to the series archive page.
1. Note that the workshops are listed chronologically, oldest first, and that there is no featured workshop.
1. Anything not working right? Anything missing?
